melodies_monet/plots/aircraftplots.py

```diff
@@ -0,0 +1,538 @@
+        latmin,lonmin,latmax,lonmax,acro = get_epa_bounds(index=None,acronym=domain_name)
+        plt.title('EPA Region ' + domain_name + ': ' + label_m + ' - ' + label_o,fontweight='bold',**text_kwargs)
+    else:
+        latmin= math.floor(min(df.latitude))
+        lonmin= math.floor(min(df.longitude))
+        latmax= math.ceil(max(df.latitude))
+        lonmax= math.ceil(max(df.longitude))
+        plt.title(domain_name + ': ' + label_m + ' - ' + label_o,fontweight='bold',**text_kwargs)
+    
+    if 'extent' not in map_kwargs:
+        map_kwargs['extent'] = [lonmin,lonmax,latmin,latmax]  
+    ax.axes.set_extent(map_kwargs['extent'],crs=ccrs.PlateCarree())
+    
+    #Update colorbar
+    f = plt.gcf()
+    model_ax = f.get_axes()[0]
+    cax = f.get_axes()[1]
+    #get the position of the plot axis and use this to rescale nicely the color bar to the height of the plot.
+    position_m = model_ax.get_position()
+    position_c = cax.get_position()
+    cax.set_position([position_c.x0, position_m.y0, position_c.x1 - position_c.x0, (position_m.y1-position_m.y0)*1.1])
+    cax.set_ylabel(ylabel,fontweight='bold',**text_kwargs)
+    cax.tick_params(labelsize=text_kwargs['fontsize']*0.8,length=10.0,width=2.0,grid_linewidth=2.0)    
+    
+    #plt.tight_layout(pad=0)
+    savefig(outname + '.png', loc=4, logo_height=120)
+    
+####NEW function for adding 'altitude' variable as secondary y- axis (qzr++)
+def add_yax2_altitude(ax, pairdf, altitude_yax2, text_kwargs): 
+
+    """Creates secondary y-axis (altitude) for timeseries plot.
+    
+    Parameters
+    ----------
+    ax : ax
+        Matplotlib ax from previous occurrences so it can overlay obs and model results on the same plot.
+    pairdf : pandas.DataFrame
+        Model/obs paired data to plot.
+    text_kwargs : dictionary
+        Dictionary containing information about text.
+    altitude_yax2: dictionary
+        Secondary y-axis (altitude) control options, including altitude_variable, altitude_ticks, etc.
+                
+    Returns
+    -------
+    ax : ax
+        Matplotlib ax such that driver.py can iterate to overlay multiple models on the same plot.
+    """
+    ax2 = ax.twinx()
+    
+    # Fetch altitude parameters from altitude_yax2
+    altitude_variable = altitude_yax2['altitude_variable']
+    altitude_ticks = altitude_yax2['altitude_ticks']
+    plot_kwargs_y2 = altitude_yax2.get('plot_kwargs_y2', {})
+    ylabel2 = altitude_yax2.get('ylabel2', 'Altitude')
+
+    # Plot altitude
+    ax2.plot(pairdf.index, pairdf[altitude_variable], **plot_kwargs_y2, label=ylabel2)
+    
+    # Set labels, ticks, and limits
+    ax2.set_ylabel(ylabel2, fontweight='bold', fontsize=text_kwargs['fontsize'], color=plot_kwargs_y2.get('color', 'g'))
+    ax2.tick_params(axis='y', labelcolor=plot_kwargs_y2.get('color', 'g'), labelsize=text_kwargs['fontsize'] * 0.8)
+    ax2.set_ylim(pairdf[altitude_variable].min(), pairdf[altitude_variable].max() + altitude_ticks)
```

Comment:
Can you make this the default but add the option for vmin_y2 and vmax_y2 in the `altitude_yax2` dict in the control yaml and if they are defined then use those to set the y limits? That provides a slightly different functionality than the filtering, such as if you wanted to have a wider range of altitude plotted than min/max.